### PR TITLE
refactor video overlay js to tsx

### PR DESCRIPTION
**Pull request by sickelap**
+ make eslint a tiny bit happier

### CODE REVIEW

File: src/components/photolist/PhotoListView.tsx [hunk 1]
```diff
@@ -32,14 +32,14 @@ type Props = {
...
-  header?: any;
-  additionalSubHeader?: any;
+  isPublic: boolean;
+  numberOfItems: number;
+  updateGroups: any;
+  updateItems: any;
+  date: any;
+  dayHeaderPrefix: any;
+  header: any;
+  additionalSubHeader: any;
```
Comment (polaroidkidd) [inline src/components/photolist/PhotoListView.tsx]: any chance we can avoid having `any` types?

File: src/components/photolist/PhotoListView.tsx [hunk 2]
```diff
@@ -87,30 +87,32 @@ function PhotoListViewComponent(props: Props) {
     idx2hashRef.current = idx2hash;
   }, [idx2hash]);
 
+  /* eslint-disable react-hooks/exhaustive-deps */
```
Comment (polaroidkidd) [inline src/components/photolist/PhotoListView.tsx:90]: why disable this linting rule here and on line 96?

File: src/components/photolist/PhotoListView.tsx [hunk 3]
```diff
@@ -87,30 +87,32 @@ function PhotoListViewComponent(props: Props) {
...
-  const getUrl = useCallback((url, pxHeight) => {
+  const getUrl = useCallback((url: string, pxHeight: number) => {
     if (pxHeight < 250) {
       return `${serverAddress}/media/square_thumbnails_small/${url.split(";")[0]}`;
     }
     return `${serverAddress}/media/square_thumbnails/${url.split(";")[0]}`;
   }, []);
 
-  const updateSelectionState = newState => {
+  const updateSelectionState = (newState: { selectedItems: any[]; selectMode: boolean }) => {
```
Comment (polaroidkidd) [inline src/components/photolist/PhotoListView.tsx:109]: if we're not sure what `selectedItems` will be at the time of invocation, we might want to think about typing it as `unknown`.

File: src/components/photolist/PhotoListView.tsx [hunk 4]
```diff
@@ -140,7 +142,7 @@ function PhotoListViewComponent(props: Props) {
     });
   };
 
-  const handleClick = (event, item) => {
+  const handleClick = (event: { shiftKey: any }, item: { id: any }) => {
```
Comment (polaroidkidd) [inline src/components/photolist/PhotoListView.tsx]: the event should be `React.KeyboardEvent<SOME HTML ELEMENT>` and the `id` should have a consistent type as well, so there shouldn't the be need to type them as `any`.

File: src/components/photolist/PhotoListView.tsx [hunk 5]
```diff
@@ -140,7 +142,7 @@ function PhotoListViewComponent(props: Props) {
...
     // if an image is selectable, then handle shift click
     if (selectable && event.shiftKey) {
       const lastSelectedElement = selectionStateRef.current.selectedItems.slice(-1)[0];
```
Comment (polaroidkidd) [inline src/components/photolist/PhotoListView.tsx]: could be not use `selectionStateRef.current.selectedItems.at(-1)` to get the last selected item?

File: src/components/photolist/PhotoListView.tsx (hunk 1 above)
Comment (sickelap) [reply src/components/photolist/PhotoListView.tsx]: I'm not happy about this either. To make it a simple change I left them as they are. I plan to review the application state model and when that's done will look into site-wide type usages later.

File: src/components/photolist/PhotoListView.tsx (hunk 2 above)
Comment (sickelap) [reply src/components/photolist/PhotoListView.tsx:90]: I wasn't sure how to address this when eslint was complaining about react hook dependencies not being provided.

If you have an idea how to address this I'd be happy to learn :)

File: src/components/photolist/PhotoListView.tsx (hunk 3 above)
Comment (sickelap) [reply src/components/photolist/PhotoListView.tsx:109]: I think I tried to use `unknown` but then propagated types went nuts. This was to make eslint happy. I did not put much effort to fix types properly. As I mentioned above I'll address site-wide types later.

File: src/components/photolist/PhotoListView.tsx (hunk 4 above)
Comment (sickelap) [reply src/components/photolist/PhotoListView.tsx]: I completely agree. I did not want to pass `-n` to commit command to ignore pre-commit checks hence some types to make linter happy. This also highlights the fact that something needs to be done about types in the app.
When refactoring site-wide types (I hope) all `any`s will be gone.

File: src/components/photolist/PhotoListView.tsx (hunk 5 above)
Comment (sickelap) [reply src/components/photolist/PhotoListView.tsx]: I'd be more than happy to use modern techniques but according to [this](https://github.com/LibrePhotos/librephotos-frontend/blob/dev/package.json#L32) we support IE11. And according to [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at#browser_compatibility) `Array.prototype.at` does not work on IE11.

Comment (polaroidkidd) [reply src/components/photolist/PhotoListView.tsx]: oh my.. I didn't see that. @derneuere   do we really support IE11? It's gone out of support, so I don't really see a reason why we should support it..

Comment (derneuere) [reply src/components/photolist/PhotoListView.tsx]: I think it makes sense to drop IE11 support :)

Comment (sickelap) [reply src/components/photolist/PhotoListView.tsx]: done

File: src/components/photolist/PhotoListView.tsx (hunk 1 above)
Comment (derneuere) [reply src/components/photolist/PhotoListView.tsx]: These attributes are all optional. I am not sure, why you changed them. If I check out your pull request, I got a lot of errors within the different layout pages.

I think reverting these changes will be enough to merge the rest :)

Comment (sickelap) [reply src/components/photolist/PhotoListView.tsx]: my bad, I did not check this properly. I'm reverting this.

File: src/components/photolist/PhotoListView.tsx [hunk 6]
```diff
@@ -374,6 +376,17 @@ function PhotoListViewComponent(props: Props) {
...
+PhotoListViewComponent.defaultProps = {
+  isPublic: null,
+  numberOfItems: null,
+  updateItems: null,
+  date: null,
+  dayHeaderPrefix: null,
+  header: null,
+  additionalSubHeader: null,
+  updateGroups: null,
+};
```
Comment (sickelap) [inline src/components/photolist/PhotoListView.tsx:388]: These are necessary to make eslint happy (rule `react/require-default-props`)

File: src/components/photolist/PhotoListView.tsx (hunk 4 above)
Comment (sickelap) [reply src/components/photolist/PhotoListView.tsx]: updated types

**Comment by derneuere** [comment]
Looks good to me :+1:

**Pull request merged**